src/main/webapp/service/timesheet.service.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+angular
+    .module('sharedTimesheetApp')
+    .service('timesheetService', timesheetService);
+
+timesheetService.$inject = ['$http', '$q'];
+
+function timesheetService($http, $q) {
```

Comment:
consider refactoring this service using promise syntax, you can completely avoid using $q defer library dependency.